tests/drivers/can/api/src/main.c

```diff
@@ -996,7 +982,19 @@ static void test_filters_preserved_through_bitrate_change(void)
 	can_remove_rx_filter(can_dev, filter_id);
 }
 
-void test_main(void)
+/**
+ * set can mode to loopback mode before sending/receiving test cases as it
+ * allows these test cases to send/receive their own frames.
+ */
+void set_loopback(void)
+{
+	int err;
+
+	err = can_set_mode(can_dev, CAN_MODE_LOOPBACK);
+	zassert_equal(err, 0, "failed to set loopback-mode (err %d)", err);
+}
+
```

Comment:
Please just roll this into the `setup` function and convert it to using `zassume` instead of `zassert` (as `zassert` should not be called from the setup function).